lib/mongo/bulk_write/validatable.rb

```diff
@@ -43,6 +43,10 @@ def validate(name, document)
         if document.respond_to?(:keys) && (document[:collation] || document[Operation::COLLATION])
           @has_collation = true
         end
+
+        if document.respond_to?(:keys) && (document[:array_filters] || document[Operation::ARRAY_FILTERS])
```

Comment:
You only need to check if the document has the key ``:array_filters`` or ``'array_filters'``

Because I've suggested changing ``Operation::ARRAY_FILTERS`` to ``arrayFilters`` below, you shouldn't use this constant.